includes/class-sensei-course.php

```diff
@@ -1264,6 +1264,62 @@ public function course_lessons( $course_id = 0, $post_status = 'publish', $field
 		return $lessons;
 
 	}
+	/**
+	 * Get lessons by courses.
+	 *
+	 * @since  4.3.0
+	 * @access public
+	 *
+	 * @param array $courses_ids The list of courses ids.
+	 * @return array[] lists of lessons by courses.
+	 */
+	public function get_lessons_grouped_by_courses( $courses_ids ) {
```

Comment:
This method looks very specific to reports. I'm not sure if we need it here. Another issue might be that it's public which makes it hard to refactor later.

Same could be said about `Sensei_Course::get_students_grouped_by_courses` and `Sensei_Lesson::get_completed_lessons_for_students_and_lessons`

Reply:
When I see a method name like this I expect the return results to be lessons grouped by courses. Something like:

```php
$lessons = [
	1 => [ 2, 3 ], // Where the key is the course id and the inner array contains the lesson ids.
];
```
I think here you are using the courses more like a filter, but you are grouping the courses in `Sensei_Analysis_Overview_List_Table::get_lessons_by_courses`, so maybe combine the methods and rename the second one?

Same applies for `get_students_grouped_by_courses`.